segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java

```diff
@@ -257,4 +257,31 @@ public static void main(String[] args) throws Exception {
     }
 
     //endregion
+
+    /**
+     * This custom factory is used to ensure that Zookeeper clients in Curator are always created using the Zookeeper
+     * hostname, so it can be resolved to a new IP in the case of a Zookeeper instance restart.
+     */
+    static class ZKClientFactory implements ZookeeperFactory {
```

Comment:
Make this class Thread safe. Mark it as such and annotate the member variables with the GuardedBy

Reply:
Done.